temporalio/exceptions.py

```diff
@@ -23,7 +41,17 @@ def cause(self) -> Optional[BaseException]:
 
 
 class FailureError(TemporalError):
-    """Base for runtime failures during workflow/activity execution."""
+    """Base class for exceptions that cause a workflow execution failure.
+
+    Do not raise this directly in workflow code: raise `ApplicationError` instead (which inherits from this `FailureError`).
+
+    Workflow execution failure puts the workflow execution into "Failed" state, and no further attempts will
+    be made to progress the workflow execution.
+
+    By default, any exception that does not extend this class causes the Workflow Task to be retried, rather than failing the workflow execution.
+
+    The default behavior can be changed by providing a list of exception types to ``workflow_failure_exception_types`` when creating a worker or ``failure_exception_types`` on the ``@workflow.defn`` decorator.
+    """
```

Comment:
Formatting to 80-chars and editing tweaks:

```suggestion
    """Base class for exceptions that cause a workflow execution failure.

    Do not raise this directly: raise ``ApplicationError`` instead.

    Workflow execution failure puts the workflow execution into "Failed" state,
    and no further attempts will be made to progress the workflow execution.

    By default, any exception that does not inherit from this class causes the
    workflow task to be retried, rather than failing the workflow execution. The
    default behavior can be changed by providing a list of exception types to
    ``workflow_failure_exception_types`` when creating a worker or
    ``failure_exception_types`` on the ``@workflow.defn`` decorator.
    """
```